functions/memorystore/redis/visitcount.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START functions_memorystore_redis]
+
+// Package visitcount provides a Cloud Function that connects
+// to a managed Redis instance.
+package visitcount
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/gomodule/redigo/redis"
+)
+
+var redisPool *redis.Pool
+
+// Initialize the connection pool on instance startup
+func init() {
```

Comment:
I updated #877 to include testing this. For now, please change this to `func initializeRedis() (*redis.Pool, error)` and have it return errors (`return nil, fmt.Errorf("...", err)`).

Then, in `VisitCount`, you can do:

```go
if redisPool == nil {
	// Pre-declare err to avoid shadowing redisPool.
	var err error
	redisPool, err = initializeRedis()
	if err != nil {
		log.Fatalf("initializeRedis: %v", err)
	}
	// ...
}
```

This makes writing to the global `redisPool` variable explicit - it happens at the beginning of `VisitCount`.

It also makes it easier to test because you don't initialize the pool until you run the function. That means a test can check the environment variables are set before testing the function -- ideally when someone clones this repo and runs `go test`, everything would pass.

Hopefully we won't have to modify the sample code at all when we write the test.

Reply:
Done - let me know if the rewrite looks okay!